File: llm_generated_code/ChatGPT/CPP/Level2/Q37.cpp
#include <iostream>
#include <string>
using namespace std;

int min_coins_to_transform(const string& s) {
    int count_0 = 0, count_1 = 0;
    for (char c : s) {
        if (c == '0') count_0++;
        else count_1++;
    }
    return min(count_0, count_1);
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;
        cout << min_coins_to_transform(s) << endl;
    }
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level2/Q37.py
def main():
    for _ in range(int(input())):
        s = input()
        n = len(s)
        count_0 = s.count('0')
        count_1 = s.count('1')
        result = min(count_0, count_1)
        print(result)

if __name__ == "__main__":
    main()
